docs/2_2_1_fingerprinting.md

```diff
@@ -194,77 +196,89 @@ val patch = bytecodePatch {
 > If the fingerprint can not be matched to any method, the match of a fingerprint is `null`. If such a match is delegated
 > to a variable, accessing it will raise an exception.
 
-The match of a fingerprint contains mutable and immutable references to the method and the class it matches to.
+> [!TIP]
+> If a fingerprint has an opcode pattern, you can use the `fuzzyPatternScanThreshhold` parameter of the `opcode`
+> function to fuzzy match the pattern.  
+> `null` can be used as a wildcard to match any opcode:
+>
+> ```kt
+> fingerprint(fuzzyPatternScanThreshhold = 2) {
+>   opcodes(
+>     Opcode.ICONST_0,
+>     null,
+>     Opcode.ICONST_1,
+>     Opcode.IRETURN,
+>    )
+>}
+> ```
+> 
+The match of a fingerprint contains references to the original method and class definition of the method:
 
 ```kt
 class Match(
-    val method: Method,
-    val classDef: ClassDef,
+    val originalMethod: Method,
+    val originalClassDef: ClassDef,
     val patternMatch: Match.PatternMatch?,
     val stringMatches: List<Match.StringMatch>?,
     // ...
 ) {
-    val mutableClass by lazy { /* ... */ }
-    val mutableMethod by lazy { /* ... */ }
+    val classDef by lazy { /* ... */ }
+    val method by lazy { /* ... */ }
```

Comment:
I think in the source that's exactly what I did. The reason I swapped mutableMethod with method was because the former was being used say more often, so it makes sense to use a shorter name. Additionally I renamed the immutable method originalMethod because I am not really sure if we want to continue with "mutable" and "immutable". In fact the entire reason this exists is because DexLib has this Immutable concept, but there's not really any reason not to add setters. Doing so would allow us to get rid of these mutable copy classes as we could simply ser fields in the classes.

Reply:
A comment already exists in the source:

![image](https://github.com/user-attachments/assets/c4a54477-9387-4178-8b03-f0d71c72a864)

I'll add it to the docs too.